src/inputs.rs

```diff
@@ -93,7 +101,10 @@ impl<B: Backend> CompiledCircuit<B> {
 
             (TyKind::Array(el_typ, size), Value::Array(values)) => {
                 if values.len() != (*size as usize) {
-                    panic!("wrong size of array");
+                    Err(ParsingError::ArraySizeMismatch(
+                        values.len(),
+                        *size as usize,
+                    ))?
```

Comment:
in general we should not replace panics with specific errors as we don't know if we'll ever reach them. Specific errors are good if we know the user can reach them by writing bad code. In this case though, we clearly have a reachable error so it's a good change!